es/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -143,6 +144,19 @@
      */
     static final Setting<String> BASE_PATH_SETTING = Setting.simpleString("base_path");
 
+    public static List<Setting> optionalSettingsToValidate() {
```

Comment:
why those are called optionalSettings?

Reply:
These are considered optional for the parameters provided at the `create repository` statement. 
See https://github.com/crate/crate/blob/master/sql/src/main/java/io/crate/analyze/repositories/TypeSettings.java#L44

I'm open to other suggestions.

Reply:
i was asking it coz the settings list has ACCESS_KEY_SETTING and SECRET_KEY_SETTING, but i see